src/ui/widgets/Readback/readback.tsx

```diff
@@ -108,42 +124,115 @@ export const ReadbackComponent = (
-        foregroundColor = new Color("var(--alarm)");
-        break;
+      case AlarmQuality.WARNING:
+        foregroundColor = "var(--alarm)";
+        border = "2px solid var(--alarm)";
     }
   }
-  // Use a LabelComponent to display it.
+
+  const font = props.font?.css() ?? diamondTheme.typography;
+
+  const backgroundColor = transparent
+    ? "transparent"
+    : (props.backgroundColor?.toString() ?? diamondTheme.palette.primary.main);
+
+  let alignmentV = "center";
+  if (textAlignV === "top") {
+    alignmentV = "start";
+  } else if (textAlignV === "bottom") {
+    alignmentV = "end";
+  }
+
+  // If props.font exists, extracts the font size in rem and returns is back to size in px
+  // using the default browser size of 16px, as used in ../../../types/font.ts
+  const fontSize = props.font?.css().fontSize
+    ? parseFloat(
+        props.font
+          .css()
+          .fontSize?.toString()
+          .match(/\d+.\d+/)
+          ?.toString() ?? ""
+      ) * 16
+    : diamondTheme.typography.fontSize;
+
+  const inputWidth = rotationStep === 0 || rotationStep === 2 ? width : height;
+  const inputHeight = rotationStep === 0 || rotationStep === 2 ? height : width;
+
+  const offset = width / 2 - height / 2;
+  const transform =
+    rotationStep === 1 || rotationStep === 3
+      ? `rotate(${rotationStep * -90}deg) translateY(${offset}px) translateX(${offset}px)`
+      : `rotate(${rotationStep * -90}deg)`;
+
+  // Calculate max number of rows based on the height of the widget and the height of the font
+  // an extra row is then subtracted to make it fit nicer
+  const maxRows =
+    Math.floor(inputHeight / fontSize) - 1 < 1
+      ? 1
+      : Math.floor(inputHeight / fontSize) - 1;
+
   return (
-    <LabelComponent
-      className={className}
-      text={displayedValue}
-      transparent={transparent}
-      textAlign={textAlign}
-      textAlignV={textAlignV}
-      font={font}
-      foregroundColor={foregroundColor}
-      backgroundColor={backgroundColor}
-      border={border}
-      rotationStep={rotationStep}
-      visible={visible}
-      wrapWords={wrapWords}
-    ></LabelComponent>
+    <TextField
+      disabled={!enabled}
+      value={displayedValue}
+      multiline={wrapWords}
+      maxRows={maxRows}
+      variant="outlined"
+      slotProps={{
+        input: {
+          readOnly: true
+        }
+      }}
+      sx={{
+        "&.MuiFormControl-root": {
+          height: inputHeight,
+          width: inputWidth,
+          display: "flex",
+          transform: transform
+        },
+        "& .MuiInputBase-input": {
+          textAlign: textAlign,
+          font: font,
+          lineHeight: 1
+        },
+        "& .MuiInputBase-root": {
+          alignItems: alignmentV,
+          color: foregroundColor,
+          backgroundColor: backgroundColor
+        },
+        "& .MuiOutlinedInput-root": {
+          "& .MuiOutlinedInput-notchedOutline": {
+            border: border,
```

Comment:
I'm noticing that when there's a border present that's more than a few pixels thick it seems to cut off some of the text inside the box. This seems like an MUI feature but it does mean it's hard to read some text displayed (in comparison, Phoebus has the border expand outside the width and height of the component). It would be good to explore if this can be fixed.
![Screenshot from 2025-04-23 15-44-31](https://github.com/user-attachments/assets/b83d1b00-2943-4a30-bd31-6c2542bb89bf)